packages/valory/skills/ceramic_read_abci/behaviours.py

```diff
@@ -0,0 +1,147 @@
+
+    @property
+    def synchronized_data(self) -> SynchronizedData:
+        """Return the synchronized data."""
+        return cast(SynchronizedData, super().synchronized_data)
+
+    @property
+    def params(self) -> Params:
+        """Return the params."""
+        return cast(Params, super().params)
+
+    def _get_stream_data(self, stream_id: str) -> Generator[None, None, Optional[dict]]:
+        """Get the current data from a Ceramic stream"""
+
+        api_base = self.params.ceramic_api_base
+        api_endpoint = self.params.ceramic_api_read_endpoint
+        url = api_base + api_endpoint.replace("{stream_id}", stream_id)
+
+        self.context.logger.info(f"Reading data from Ceramic API [{url}]")
+        response = yield from self.get_http_response(
+            method="GET",
+            url=url,
+        )
+
+        if response.status_code != 200:
+            self.context.logger.error(
+                f"API error while reading the stream: {response.status_code}: '{response.body!r}'"
+            )
+            return None
+
+        try:
+            api_data = json.loads(response.body)
+        except json.decoder.JSONDecodeError:
+            self.context.logger.error(
+                f"API error while loading the response json. Response body: '{response.body!r}'"
+            )
+            return None
+
+        # Extract first and last commit info
+        genesis_cid_str = api_data["commits"][0]["cid"]
+        previous_cid_str = api_data["commits"][-1]["cid"]
+
+        # Rebuild the current data
+        self.context.logger.info(
+            f"Bulding stream data from commits:\n'{api_data['commits']!r}'"
+        )
+        data = build_data_from_commits(api_data["commits"])
+
+        self.context.logger.info(f"Got data from Ceramic API: {data}")
+
+        return {
+            "genesis_cid_str": genesis_cid_str,
+            "previous_cid_str": previous_cid_str,
+            "data": data,
+        }
+
+class StreamReadBehaviour(CeramicReadBaseBehaviour):
+    """StreamReadBehaviour"""
+
+    matching_round: Type[AbstractRound] = StreamReadRound
+
+    def async_act(self) -> Generator:
+        """Do the act, supporting asynchronous execution."""
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+
+            # stream_id and target_property_name can be set either in the synchronized_data or as a param. The former has higher priority.
+            stream_id = self.synchronized_data.stream_id or self.params.stream_id
+            if not stream_id:
+                raise ValueError("stream_id has not been set neither in the synchronized_data nor as a parameter")
+
+            target_property_name = self.synchronized_data.target_property_name or self.params.target_property_name
+            if not target_property_name:
+                raise ValueError("target_property_name has not been set neither in the synchronized_data nor as a parameter")
```

Comment:
Are we ok with this approach? This skill reads a stream_id that can be configured either in the synchronized_data or in a param.

Reply:
So if its not in sync data that means that this agent has a bad state. Not sure what's the point for having `self.params` as a backup, if it is not synced across the agents.

Reply:
Yes, so the idea behind this extraction is to have a skill that loads the content of a Ceramic stream into synchronized_data. There are two possible cases that I'm interested i:
- Load a stream in the middle of the service execution: agents dynamically set the stream_id and also the target_property_name, so the content on the stream is loaded there and the skill can be re-used for different streams loaded into different synchronized_data properties.
- Load a stream at the beginning of the service execution: here this skill is the initial one, so agents havent set any stream_id or target_property_name. The only way is to use params. Or maybe we can set the initial db values?

Reply:
Okay I see what you mean now. So at startup only the params version is available. I don't see anything wrong with the way it currently is. One possible issue that might arise, is if the agent instances have this param set differently, with the way that it currently is, we would understand that is the case when this round would be executed.

Reply:
Correct, so in order for that to happen (the NO_MAJORITY case), the execution would need to reach this round.

Reply:
I've confirmed with Manos that we dont support values in the synced data being populated after registration.

Reply:
I am sorry, I did not have the full context. Now that I studied the PR and the comments above, I understand that all you need is to add this in the `setup` of the configuration.

Moreover, you could also override the `_ensure_setup` in `BaseParams` in order to ensure that the key is always set.

We could do this on a separate PR after the stacked PRs you have prepared get into `main`.